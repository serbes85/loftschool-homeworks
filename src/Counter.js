import React, { useState } from "react";

export const Counter = () => {
  const [count, setCount] = useState(0);
  const increment = () => {
    setCount((count) => count + 1);
  };

  const decrement = () => {
    setCount((count) => count - 1);
  };

  return (
    <>
      <div>{count}</div>
      <button className="btn-increment" onClick={increment}>
        Increment
      </button>
      <button className="btn-decrement" onClick={decrement}>
        Decrement
      </button>
    </>
  );
};
