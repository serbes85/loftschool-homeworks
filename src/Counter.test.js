// Компонент Counter - простой счётчик
// Напишите для него тесты.
// Убедитесь, что вы протестировали всю функциональнось.
// Также проверьте что компонент рендерится верно, используя Snapshot тест.

// * Задание со звёздочкой - выполнять не обязательно

// Вынесите логику в хук и протестируйте его
import React from "react";
import { Counter } from "./Counter";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

it("increment", () => {
  const wrapper = shallow(<Counter />);

  expect(wrapper.find("div").text()).toEqual("0");
  wrapper.find(".btn-increment").simulate("click");
  expect(wrapper.find("div").text()).toEqual("1");
});
it("is value negative", () => {
  const wrapper = shallow(<Counter />);

  wrapper.find(".btn-decrement").simulate("click");
  expect(wrapper.find("div").text()).toEqual("-1");
});
it("decrement", () => {
  const wrapper = shallow(<Counter />);

  expect(wrapper.find("div").text()).toEqual("0");
  wrapper.find(".btn-decrement").simulate("click");
  expect(wrapper.find("div").text()).toEqual("-1");
});

it("renders correctly", () => {
  const tree = renderer.create(<Counter />).toJSON();

  expect(tree).toMatchSnapshot();
});
