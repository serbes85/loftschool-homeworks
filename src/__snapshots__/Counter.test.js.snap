// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
Array [
  <div>
    0
  </div>,
  <button
    className="btn-increment"
    onClick={[Function]}
  >
    Increment
  </button>,
  <button
    className="btn-decrement"
    onClick={[Function]}
  >
    Decrement
  </button>,
]
`;
